Documenting, if I flip the default value of the setting (current allows, switch to not-allow), that results in the "Consumers" checks failing

### eda-server

```
ERROR tests/integration/api/test_user.py::test_create_superuser - rest_framework.exceptions.ValidationError: [ErrorDetail(string="{'permissions', 'Local custom roles can only include view permission for shared models'}", code='invalid')]
ERROR tests/integration/api/test_user.py::test_modify_superuser_as_org_admin - rest_framework.exceptions.ValidationError: [ErrorDetail(string="{'permissions', 'Local custom roles can only include view permission for shared models'}", code='invalid')]
ERROR tests/integration/api/test_user.py::test_organization_admin_can_create_user - rest_framework.exceptions.ValidationError: [ErrorDetail(string="{'permissions', 'Local custom roles can only include view permission for shared models'}", code='invalid')]
```

### AWX

```
FAILED awx/main/tests/functional/test_rbac_execution_environment.py::test_give_object_permission_to_ee
FAILED awx/main/tests/functional/test_rbac_execution_environment.py::test_give_org_permission_to_ee[new]
FAILED awx/main/tests/functional/test_rbac_execution_environment.py::test_give_org_permission_to_ee[old]
```

```
  File "/awx_devel/awx/main/tests/functional/test_rbac_execution_environment.py", line 136, in test_give_org_permission_to_ee
    check_user_capabilities(org_member, org_ee, {'edit': False, 'delete': False, 'copy': False})
  File "/awx_devel/awx/main/tests/functional/test_rbac_execution_environment.py", line 58, in _rf
    raise RuntimeError(f'Could not find expected object ({obj}) in EE list result: {r.data}')
RuntimeError: Could not find expected object (some user ee-1) in EE list result: {'count': 0, 'next': None, 'previous': None, 'results': []}
```

```
  File "/awx_devel/awx/main/tests/functional/test_rbac_execution_environment.py", line 103, in test_give_object_permission_to_ee
    assert access.can_read(org_ee)  # by virtue of being an org member
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: assert False
 +  where False = can_read(<ExecutionEnvironment: some user ee-1>)
 +    where can_read = <awx.main.access.ExecutionEnvironmentAccess object at 0x7f0530f6e010>.can_read
```

So that looks like this breaks some rule related to visibility of EEs.

--- a/ansible_base/rbac/validators.py
+++ b/ansible_base/rbac/validators.py
@@ -156,6 +156,17 @@ def validate_permissions_for_model(permissions, content_type: Optional[Model], m
     if settings.ANSIBLE_BASE_DELETE_REQUIRE_CHANGE and role_model is not None:
         check_has_change_with_delete(codename_set, permissions_by_model)
 
+    if not managed:
+        for perm in permissions:
+            # View permission for shared objects is interpreted as permission to view
+            # the resource locally, which is needed to be able to view parent objects
+            # For system roles, this exception is not allowed
+            if content_type and perm.codename.startswith('view'):
+                continue
+            model = perm.content_type.model_class()
+            if permission_registry.get_resource_prefix(model) == 'shared':
+                raise ValidationError({'permissions', 'Not managed locally, local custom roles can only include view'})
+
 
 def validate_codename_for_model(codename: str, model: Union[Model, Type[Model]]) -> str:
     """Shortcut method and validation to allow action name, codename, or app_name.codename

--- a/test_app/tests/rbac/api/test_rbac_validation.py
+++ b/test_app/tests/rbac/api/test_rbac_validation.py
@@ -36,19 +36,23 @@ class TestSharedAssignmentsDisabled:
         assert response.status_code == 201, response.data
         assert rando.has_obj_perm(team, 'member')
 
-    @override_settings(ALLOW_SHARED_RESOURCE_CUSTOM_ROLES=False)
-    def test_custom_roles_for_shared_stuff_not_allowed(self, admin_api_client):
+    @pytest.mark.parametrize('allowed', [True, False])
+    def test_custom_roles_for_shared_stuff_not_allowed(self, admin_api_client, allowed):
         url = get_relative_url('roledefinition-list')
-        response = admin_api_client.post(
-            url,
-            data={
-                'name': 'Alternative Organization Admin Role in Local Server',
-                'content_type': 'aap.organization',
-                'permissions': ['aap.view_organization', 'local.change_organization'],
-            },
-        )
-        assert response.status_code == 400, response.data
-        assert 'Local custom roles can only include view permission for shared models' in str(response.data)
+        with override_settings(ALLOW_SHARED_RESOURCE_CUSTOM_ROLES=allowed):
+            response = admin_api_client.post(
+                url,
+                data={
+                    'name': 'Alternative Organization Admin Role in Local Server',
+                    'content_type': 'aap.organization',
+                    'permissions': ['aap.view_organization', 'local.change_organization'],
+                },
+            )
+        if allowed is False:
+            assert response.status_code == 400, response.data
+            assert 'Local custom roles can only include view permission for shared models' in str(response.data)
+        else:
+            assert response.status_code == 201, response.data
 
     @override_settings(ALLOW_LOCAL_RESOURCE_MANAGEMENT=False)
     def test_auditor_for_external_models(self, admin_api_client, rando, external_auditor_constructor):

--- a/test_app/tests/rbac/api/test_rbac_validation.py
+++ b/test_app/tests/rbac/api/test_rbac_validation.py
@@ -23,7 +23,7 @@ class TestSharedAssignmentsDisabled:
         assert response.status_code == 400, response.data
         assert self.NON_LOCAL_MESSAGE in str(response.data)
 
-    @override_settings(ALLOW_LOCAL_RESOURCE_MANAGEMENT=False)
+    @override_settings(ALLOW_SHARED_RESOURCE_CUSTOM_ROLES=False)
     def test_custom_roles_for_shared_stuff_not_allowed(self, admin_api_client):
         url = get_relative_url('roledefinition-list')
         response = admin_api_client.post(
@@ -35,7 +35,7 @@ class TestSharedAssignmentsDisabled:
             },
         )
         assert response.status_code == 400, response.data
-        assert self.NON_LOCAL_MESSAGE in str(response.data)
+        assert 'Local custom roles can only include view permission for shared models' in str(response.data)
 
     @override_settings(ALLOW_LOCAL_RESOURCE_MANAGEMENT=False)
     def test_auditor_for_external_models(self, admin_api_client, rando, external_auditor_constructor):

